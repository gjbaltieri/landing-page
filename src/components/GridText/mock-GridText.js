// eslint-disable-next-line import/no-anonymous-default-export
export default {
  background: false,
  title: 'meu',
  description: 'description',
  grid: [
    {
      title: 'Title 1',
      description:
        'Dolore magna aliqua. Faucibus interdum posuere lorem ipsum dolor. Libero volutpat sed cras ornare arcu dui vivamus arcu felis. Auctor augue mauris augue neque gravida in. Est placerat in egestas erat imperdiet. Habitant morbi tristique senectus et netus. Congue nisi vitae suscipit tellus mauris. Luctus venenatis lectus magna fringilla urna porttitor rhoncus dolor. Scelerisque felis imperdiet proin fermentum leo vel orci. Pellentesque sit amet porttitor eget dolor morbi. Semper risus in hendrerit gravida rutrum quisque non.',
    },

    {
      title: 'Title 2',
      description:
        'Amet risus nullam eget felis eget nunc. Sed euismod nisi porta lorem mollis aliquam ut porttitor. Vel facilisis volutpat est velit egestas dui id. In nibh mauris cursus mattis molestie a iaculis at. Cras adipiscing enim eu turpis egestas pretium. Interdum varius sit amet mattis vulputate enim nulla aliquet porttitor.',
    },

    {
      title: 'Title 3',
      description:
        'Pellentesque diam volutpat commodo sed egestas egestas fringilla phasellus faucibus. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Augue mauris augue neque gravida in fermentum et. Dolor sed viverra ipsum nunc aliquet. Dignissim cras tincidunt lobortis feugiat vivamus at augue eget arcu. Amet nulla facilisi morbi tempus iaculis urna id. ',
    },
  ],
};
