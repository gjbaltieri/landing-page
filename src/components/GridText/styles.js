import Styled, { css } from 'styled-components';
import { Title as HeadingComponent } from '../Header/Title';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = Styled.div`
${({ theme }) => css`
  ${TextComponent} {
    margin-bottom: ${theme.padding.xhuge};
  }
`}`;

export const Grid = Styled.div`
${({ theme }) => css`
  counter-reset: element-counter;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(280px, 1fr));
  overflow: hidden;
  width: 100%;
  @media ${theme.media.maxWidth768px} {
    grid-template-columns: 1fr;
  }
`}
`;

export const GridElement = Styled.div`
${({ theme }) => css`
  padding: ${theme.padding.base};
  ${HeadingComponent} {
    position: relative;
    left: 5rem;
  }

  ${HeadingComponent}::before {
    counter-increment: element-counter;
    content: counter(element-counter);
    position: absolute;
    top: -3rem;
    left: -5rem;
    font-size: ${theme.font.size.xxlarge};
    transform: rotate(5deg);
  }
`}`;
