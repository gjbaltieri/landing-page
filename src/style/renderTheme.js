import { render } from '@testing-library/react';
import { GlobalStyles } from './globalstyle';
import { theme } from './theme';
import { ThemeProvider } from 'styled-components';

export default function renderTheme(children) {
  return render(
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}
