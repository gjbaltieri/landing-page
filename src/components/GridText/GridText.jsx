import P from 'prop-types';
import * as Styled from './styles';
import SectionBackground from '../SectionBackground/SectionBackground';
import Heading from '../Header/Heading';
import TextComponent from '../TextComponent/TextComponent';

export default function GridText({
  title,
  description,
  grid,
  background = false,
}) {
  return (
    <SectionBackground background={background}>
      <Styled.Container>
        <Heading uppercase as='h2' fontSize='large' darkMode={background}>
          {title}
        </Heading>
        <TextComponent>{description}</TextComponent>

        <Styled.Grid>
          {grid.map((el) => (
            <Styled.GridElement key={el.title}>
              <Heading size='medium' darkMode={background} as='h3'>
                {el.title}
              </Heading>
              <TextComponent>{el.description}</TextComponent>
            </Styled.GridElement>
          ))}
        </Styled.Grid>
      </Styled.Container>
    </SectionBackground>
  );
}

GridText.propTypes = {
  background: P.bool,
  title: P.string.isRequired,
  description: P.string.isRequired,
  grid: P.arrayOf(
    P.shape({
      title: P.string.isRequired,
      description: P.string.isRequired,
    })
  ).isRequired,
};
