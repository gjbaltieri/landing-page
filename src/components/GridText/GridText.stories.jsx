import React from 'react';
import GridText from './GridText';
import Mock from './mock-GridText';

export default {
  title: 'GridText',
  component: GridText,
  args: Mock,
};

export const Template = (args) => (
  <div>
    <GridText {...args} />
  </div>
);
