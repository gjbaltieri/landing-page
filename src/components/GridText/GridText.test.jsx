import 'jest-styled-components';
import '@testing-library/jest-dom';
import renderTheme from '../../style/renderTheme';
import GridText from './GridText';
import Mock from '../GridText/mock-GridText';

describe('<GridText/>', () => {
  test('should render GridText with default values', () => {
    const { container } = renderTheme(<GridText {...Mock} />);
    expect(container).toMatchSnapshot();
  });

  test('should render GridText without background', () => {
    const { container } = renderTheme(
      <GridText {...Mock} background={undefined} />
    );
    expect(container).toMatchSnapshot();
  });
});
