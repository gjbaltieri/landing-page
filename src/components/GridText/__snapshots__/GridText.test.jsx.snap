// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText/> should render GridText with default values 1`] = `
.c3 {
  color: #333333;
  font-size: font-size:3.2rem;
  font-weight: 900;
  text-transform: uppercase;
  font-size: 3.2rem;
}

.c8 {
  color: #333333;
  font-size: font-size:1.6rem;
  font-weight: 900;
  text-transform: none;
  font-size: 1.6rem;
}

.c5 {
  font-size: 1.6rem;
}

.c1 .c4 {
  margin-bottom: 6.2rem;
}

.c6 {
  counter-reset: element-counter;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(280px,1fr));
  overflow: hidden;
  width: 100%;
}

.c7 {
  padding: 1.6rem;
}

.c7 .c2 {
  position: relative;
  left: 5rem;
}

.c7 .c2::before {
  counter-increment: element-counter;
  content: counter(element-counter);
  position: absolute;
  top: -3rem;
  left: -5rem;
  font-size: 4.8rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

.c0 {
  background: #F2F2F2;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: right;
  -webkit-justify-content: right;
  -ms-flex-pack: right;
  justify-content: right;
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2 c3"
        font-size="large"
      >
        meu
      </h2>
      <p
        class="c4 c5"
      >
        description
      </p>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <h3
            class="c2 c8"
            font-size="base"
          >
            Title 1
          </h3>
          <p
            class="c4 c5"
          >
            Dolore magna aliqua. Faucibus interdum posuere lorem ipsum dolor. Libero volutpat sed cras ornare arcu dui vivamus arcu felis. Auctor augue mauris augue neque gravida in. Est placerat in egestas erat imperdiet. Habitant morbi tristique senectus et netus. Congue nisi vitae suscipit tellus mauris. Luctus venenatis lectus magna fringilla urna porttitor rhoncus dolor. Scelerisque felis imperdiet proin fermentum leo vel orci. Pellentesque sit amet porttitor eget dolor morbi. Semper risus in hendrerit gravida rutrum quisque non.
          </p>
        </div>
        <div
          class="c7"
        >
          <h3
            class="c2 c8"
            font-size="base"
          >
            Title 2
          </h3>
          <p
            class="c4 c5"
          >
            Amet risus nullam eget felis eget nunc. Sed euismod nisi porta lorem mollis aliquam ut porttitor. Vel facilisis volutpat est velit egestas dui id. In nibh mauris cursus mattis molestie a iaculis at. Cras adipiscing enim eu turpis egestas pretium. Interdum varius sit amet mattis vulputate enim nulla aliquet porttitor.
          </p>
        </div>
        <div
          class="c7"
        >
          <h3
            class="c2 c8"
            font-size="base"
          >
            Title 3
          </h3>
          <p
            class="c4 c5"
          >
            Pellentesque diam volutpat commodo sed egestas egestas fringilla phasellus faucibus. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Augue mauris augue neque gravida in fermentum et. Dolor sed viverra ipsum nunc aliquet. Dignissim cras tincidunt lobortis feugiat vivamus at augue eget arcu. Amet nulla facilisi morbi tempus iaculis urna id. 
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridText/> should render GridText without background 1`] = `
.c3 {
  color: #333333;
  font-size: font-size:3.2rem;
  font-weight: 900;
  text-transform: uppercase;
  font-size: 3.2rem;
}

.c8 {
  color: #333333;
  font-size: font-size:1.6rem;
  font-weight: 900;
  text-transform: none;
  font-size: 1.6rem;
}

.c5 {
  font-size: 1.6rem;
}

.c1 .c4 {
  margin-bottom: 6.2rem;
}

.c6 {
  counter-reset: element-counter;
  display: grid;
  grid-template-columns: repeat(auto-fit,minmax(280px,1fr));
  overflow: hidden;
  width: 100%;
}

.c7 {
  padding: 1.6rem;
}

.c7 .c2 {
  position: relative;
  left: 5rem;
}

.c7 .c2::before {
  counter-increment: element-counter;
  content: counter(element-counter);
  position: absolute;
  top: -3rem;
  left: -5rem;
  font-size: 4.8rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

.c0 {
  background: #F2F2F2;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: right;
  -webkit-justify-content: right;
  -ms-flex-pack: right;
  justify-content: right;
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2 c3"
        font-size="large"
      >
        meu
      </h2>
      <p
        class="c4 c5"
      >
        description
      </p>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <h3
            class="c2 c8"
            font-size="base"
          >
            Title 1
          </h3>
          <p
            class="c4 c5"
          >
            Dolore magna aliqua. Faucibus interdum posuere lorem ipsum dolor. Libero volutpat sed cras ornare arcu dui vivamus arcu felis. Auctor augue mauris augue neque gravida in. Est placerat in egestas erat imperdiet. Habitant morbi tristique senectus et netus. Congue nisi vitae suscipit tellus mauris. Luctus venenatis lectus magna fringilla urna porttitor rhoncus dolor. Scelerisque felis imperdiet proin fermentum leo vel orci. Pellentesque sit amet porttitor eget dolor morbi. Semper risus in hendrerit gravida rutrum quisque non.
          </p>
        </div>
        <div
          class="c7"
        >
          <h3
            class="c2 c8"
            font-size="base"
          >
            Title 2
          </h3>
          <p
            class="c4 c5"
          >
            Amet risus nullam eget felis eget nunc. Sed euismod nisi porta lorem mollis aliquam ut porttitor. Vel facilisis volutpat est velit egestas dui id. In nibh mauris cursus mattis molestie a iaculis at. Cras adipiscing enim eu turpis egestas pretium. Interdum varius sit amet mattis vulputate enim nulla aliquet porttitor.
          </p>
        </div>
        <div
          class="c7"
        >
          <h3
            class="c2 c8"
            font-size="base"
          >
            Title 3
          </h3>
          <p
            class="c4 c5"
          >
            Pellentesque diam volutpat commodo sed egestas egestas fringilla phasellus faucibus. Massa tincidunt nunc pulvinar sapien et ligula ullamcorper. Augue mauris augue neque gravida in fermentum et. Dolor sed viverra ipsum nunc aliquet. Dignissim cras tincidunt lobortis feugiat vivamus at augue eget arcu. Amet nulla facilisi morbi tempus iaculis urna id. 
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
